### PR TITLE
Remove Sympy inheritance from printers

### LINKED ISSUE #746: Remove codegen printer inheritance from sympy
Currently all the codegen printers inherit from the sympy class `StrPrinter`. As the pyccel AST no longer contains sympy objects, this inheritance no longer adds anything to pyccel. It should be removed

### PR DESCRIPTION
**Pull request by EmilyBourne**
Remove `sympy.StrPrinter` from the inheritance tree of the printer classes. Remove objects existing due to sympy. Fixes #746

### CODE REVIEW

**Review by MenoIy — APPROVED** [review]

**Review by MenoIy — APPROVED** [review]

**Review by mostafaMamoni — APPROVED** [review]

File: pyccel/codegen/printing/codeprinter.py [hunk 1]
```diff
@@ -60,7 +53,23 @@ def doprint(self, expr, assign_to=None):
...
 
+    def _print(self, expr):
+        """Annotates the AST.
+
+        The annotation is done by finding the appropriate function _visit_X
+        for the object expr. X is the type of the object expr. If this function
+        does not exist then the method resolution order is used to search for
+        other compatible _visit_X functions. If none are found then an error is
+        raised
+        """
```
Comment (yguclu) [inline pyccel/codegen/printing/codeprinter.py:64]: Was this docstring copied from the `_visit` method in the semantic parser? Please update it

File: pyccel/codegen/printing/fcode.py [hunk 2]
```diff
@@ -175,43 +170,18 @@ class FCodePrinter(CodePrinter):
...
-        'standard': 77
-    }
-
-    _operators = {
-        'and': '.and.',
-        'or': '.or.',
-        'xor': '.neqv.',
-        'equivalent': '.eqv.',
-        'not': '.not. ',
     }
```
Comment (yguclu) [inline pyccel/codegen/printing/fcode.py:174]: I've just noticed that the C printer did not have a `tabwidth` entry in the `_default_settings` dictionary, why is that?

**Review by yguclu — CHANGES_REQUESTED** [review]

File: pyccel/codegen/printing/codeprinter.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/codegen/printing/codeprinter.py]: Guilty as charged. Fixed in e29d193

File: pyccel/codegen/printing/fcode.py (hunk 2 above)
Comment (EmilyBourne) [reply pyccel/codegen/printing/fcode.py:174]: I'm not sure. It seems that the tab width is written manually in the `indent_code` method. I have modified the method in 3615e64 and c37f7ee so c and python match the behaviour in fortran

File: pyccel/codegen/printing/codeprinter.py [hunk 3]
```diff
@@ -54,9 +54,9 @@ def doprint(self, expr, assign_to=None):
...
 
     def _print(self, expr):
-        """Annotates the AST.
+        """Print the AST node in the printer language
 
-        The annotation is done by finding the appropriate function _visit_X
+        The printing is done by finding the appropriate function _visit_X
         for the object expr. X is the type of the object expr. If this function
         does not exist then the method resolution order is used to search for
         other compatible _visit_X functions. If none are found then an error is
```
Comment (yguclu) [inline pyccel/codegen/printing/codeprinter.py]: You should still change `_visit_X` with `_print_X`... :stuck_out_tongue_winking_eye:

**Review by yguclu — CHANGES_REQUESTED** [review]

File: pyccel/codegen/printing/codeprinter.py (hunk 3 above)
Comment (EmilyBourne) [reply pyccel/codegen/printing/codeprinter.py]: 🤦 fixed in 5c72531

**Review by yguclu — APPROVED** [review]

File: pyccel/codegen/printing/codeprinter.py (hunk 3 above)
Comment (yguclu) [reply pyccel/codegen/printing/codeprinter.py]: Thanks!

**Pull request merged**